src/main/java/decodes/decoder/RegexFunction.java

```diff
@@ -0,0 +1,112 @@
+package decodes.decoder;
+
+import decodes.db.DecodesScript;
+import decodes.sql.PlatformListIO;
+import ilex.var.IFlags;
+import ilex.var.TimedVariable;
+import ilex.var.Variable;
+import oracle.xml.common.regex.xerces.Match;
+import org.slf4j.LoggerFactory;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class RegexFunction
+	extends DecodesFunction
+{
+	private final static org.slf4j.Logger log = LoggerFactory.getLogger(PlatformListIO.class);
+	private int sensorNumber;
+	private String module = "regex";
+	private String argString = null;
+
+	public RegexFunction()
+	{
+	}
+
+	@Override
+	public DecodesFunction makeCopy()
+	{
+		return new RegexFunction();
+	}
+
+	@Override
+	public String getFuncName()
+	{
+		return module;
+	}
+
+	@Override
+	public void execute(DataOperations dd, DecodedMessage decmsg) throws DecoderException
+	{
+		log.trace("Executing with args '" + argString + "'");
+		this.sensorNumber = getSensorNumber();
+
+		Pattern pattern = Pattern.compile(this.argString);
+		Matcher matcher = pattern.matcher(dd.getRawMessage().toString());
```

Comment:
this should probably track with where it is in the message vs always using the whole thing but that will make the sample location below more difficult to determine and we can probably handle it as part of future work.

Unless the user has setup a command to do so, the general idea is that the functions "consume" the text they've dealt with.